pyproject.toml

```diff
@@ -398,6 +398,15 @@ Use `Union[T, None]` instead.
 which have a similar but different semantic meaning.
 See https://github.com/vega/altair/pull/3449
 """
+"narwhals.dependencies".msg = """
+Import `dependencies` from `narwhals.stable.v1` instead.
+"""
+"narwhals.typing".msg = """
+Import `typing` from `narwhals.stable.v1` instead.
+"""
+"narwhals.dtypes".msg = """
+Import `dtypes` from `narwhals.stable.v1` instead.
+"""
```

Comment:
Perfect, thanks @MarcoGorelli 

Can't wait for this to tell me off in https://github.com/vega/altair/pull/3631/files

Reply:
@MarcoGorelli I'm really happy with this, honestly might be something you wanna suggest to do in the docs

```cmd
>>> ruff check

altair\datasets\_cache.py:8:1: TID251 `narwhals.dependencies` is banned: Import `dependencies` from `narwhals.stable.v1` instead.

  |
7 | import narwhals.stable.v1 as nw
8 | from narwhals.dependencies import get_pyarrow
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TID251
9 | from narwhals.typing import IntoDataFrameT, IntoFrameT
  |

altair\datasets\_cache.py:9:1: TID251 `narwhals.typing` is banned: Import `typing` from `narwhals.stable.v1` instead.

   |
 7 | import narwhals.stable.v1 as nw
 8 | from narwhals.dependencies import get_pyarrow
 9 | from narwhals.typing import IntoDataFrameT, IntoFrameT
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TID251
10 |
11 | from altair.datasets._typing import VERSION_LATEST
   |

altair\datasets\_loader.py:5:1: TID251 `narwhals.typing` is banned: Import `typing` from `narwhals.stable.v1` instead.

  |
3 | from typing import TYPE_CHECKING, Generic, final, overload
4 |
5 | from narwhals.typing import IntoDataFrameT, IntoFrameT
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TID251
6 |
7 | from altair.datasets._readers import _Reader, backend
  |

altair\datasets\_readers.py:33:1: TID251 `narwhals.typing` is banned: Import `typing` from `narwhals.stable.v1` instead.

   |
32 | import narwhals.stable.v1 as nw
33 | from narwhals.typing import IntoDataFrameT, IntoExpr, IntoFrameT
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TID251
34 |
35 | from altair.datasets._cache import DatasetCache
   |

tests\test_datasets.py:15:1: TID251 `narwhals.dependencies` is banned: Import `dependencies` from `narwhals.stable.v1` instead.

   |
14 |   import pytest
15 | / from narwhals.dependencies import (
16 | |     is_into_dataframe,
17 | |     is_pandas_dataframe,
18 | |     is_polars_dataframe,
19 | |     is_pyarrow_table,
20 | | )
   | |_^ TID251
21 |   from narwhals.stable import v1 as nw
   |

Found 5 errors.

```